src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexDeltaMassesGenerator.cpp

```diff
@@ -123,38 +123,66 @@ namespace OpenMS
       temp_labels.push_back("no_label");
       samples_labels_.push_back(temp_labels);
     }
-    
+
     // What kind of labelling do we have?
-    // SILAC, Leu, Dimethyl, ICPL or no labelling ??
+    // SILAC, Leu, Dimethyl, ICPL, numeric labelling or no labelling ??
 
     bool labelling_SILAC = ((labels_.find("Arg") != std::string::npos) || (labels_.find("Lys") != std::string::npos));
     bool labelling_Leu = (labels_.find("Leu") != std::string::npos);
     bool labelling_Dimethyl = (labels_.find("Dimethyl") != std::string::npos);
     bool labelling_ICPL = (labels_.find("ICPL") != std::string::npos);
+    // Check whether each label string represents a double. If yes, use these doubles as mass shifts.
+    bool labelling_numeric = true;
+    for (size_t i = 0; i < samples_labels_.size(); i++)
+    {
+      for (size_t j = 0; j < samples_labels_[i].size(); j++)
+      {
+        double mass_shift;
+        try
+        {
+          mass_shift = std::stod(samples_labels_[i][j]);
+          
+          // For numeric mass shifts, long and short label names as well as the numerical mass shift are trivial.
+          // For example, long label name ("3.1415"), short label name ("3.1415") and numerical mass shift (3.1415).
+          label_delta_mass_.insert(make_pair(samples_labels_[i][j], mass_shift));
+          label_short_long_.insert(make_pair(samples_labels_[i][j], samples_labels_[i][j]));
+          label_long_short_.insert(make_pair(samples_labels_[i][j], samples_labels_[i][j]));
```

Comment:
will there be any problem if line 147 and 148 go through but 149 throws and the three vectors are not of equal lengths?

Reply:
No. Line 143 throws the exception and 147-49 are never reached. (Just tested with debug output.)
The vectors are always of equal length.